enclave/core/sgx/exception.c

```diff
@@ -405,6 +439,13 @@ void oe_virtual_exception_dispatcher(
             td->faulting_address = exinfo->maddr;
             td->error_code = exinfo->errcd;
         }
+
+        /* Update the state here to indicate the second-level exception
+         * handler is running next. This allows both exiting and the following
+         * entering flows to skip updating the state; i.e., the second-level
+         * exception handler can run with this state. */
+        td->state = OE_TD_STATE_SECOND_LEVEL_EXCEPTION_HANDLING;
```

Comment:
Upon ERESUME, `td->state` is not cleared. This would lead to a corrupted state-machine if another exception happens after ERESUME.

Reply:
Do you refer to a nested exception (occurred inside the second-level exception handler)?
In that case, the state transition is
After ERESUME: `OE_TD_STATE_SECOND_LEVEL_EXCEPTION_HANDLING`
(AEX) -> `OE_TD_STATE_FIRST_LEVEL_EXCEPTION_HANDLING` (upon enter)
-> `OE_TD_STATE_SECOND_LEVEL_EXCEPTION_HANDLING`